Gems/Atom/Feature/Common/Code/Source/FrameCaptureSystemComponent.cpp

```diff
@@ -376,40 +376,22 @@ namespace AZ
                 return false;
             }
 
-            AZ::RPI::RenderPass* renderPass = azrtti_cast<AZ::RPI::RenderPass*>(foundPasses[0]);
-            if (renderPass)
+            AZ::RPI::Pass* pass = foundPasses[0];
+            if (pass->ReadbackAttachment(m_readback, Name(slot), useOutput))
             {
-                Name slotName = Name(slot);
-                AZ::RPI::PassAttachment* attachment = nullptr;
-                for (auto& binding : renderPass->GetAttachmentBindings())
-                {
-                    if (binding.m_name == slotName)
-                    {
-                        attachment = binding.m_attachment.get();
-                        break;
-                    }
-                }
-                if (attachment)
-                {
-                    m_state = State::Pending;
-                    m_result = FrameCaptureResult::None;
-                    SystemTickBus::Handler::BusConnect();
-                    renderPass->ReadbackAttachment(m_readback, attachment);
-                }
-                else
-                {
-                    AZ_Warning("FrameCaptureSystemComponent", false, "Failed to find attachment bound to pass [%s] slot [%s]",
-                        renderPass->GetName().GetCStr(), slotName.GetCStr());
-                    return false;
-                }
+                m_state = State::Pending;
+                m_result = FrameCaptureResult::None;
+                SystemTickBus::Handler::BusConnect();
+                return true;
             }
+            AZ_Warning("FrameCaptureSystemComponent", false, "Failed to readback the attachment bound to pass [%s] slot [%s]", pass->GetName().GetCStr(), slot.c_str());
             return true;
         }
 
         bool FrameCaptureSystemComponent::CapturePassAttachmentWithCallback(const AZStd::vector<AZStd::string>& passHierarchy, const AZStd::string& slotName
```

Comment:
nit: It'd be a little cleaner to use an enum for something like this. Even though there's only two states, having an enum of UseInput or UseOutput is clearer than true and false, and it allows you to know what's going on from the calling site instead of having to look at the function declaration.